pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -268,6 +269,18 @@ func validateNodeGroupBase(ng *NodeGroupBase, path string) error {
 		return fmt.Errorf("AMI Family %s is not supported - use one of: %s", ng.AMIFamily, strings.Join(supportedAMIFamilies(), ", "))
 	}
 
+	if ng.ContainerRuntime != nil {
+		if *ng.ContainerRuntime == ContainerRuntimeContainerD && ng.AMIFamily != NodeImageFamilyAmazonLinux2 {
+			// check if it's dockerd or containerd
+			return fmt.Errorf("%s as runtime is only support for AL2 ami family", ContainerRuntimeContainerD)
+		}
+		if *ng.ContainerRuntime != ContainerRuntimeDockerD {
+			if *ng.ContainerRuntime != ContainerRuntimeContainerD {
```

Comment:
possible to one-line this? or did you not like the length?
```suggestion
		if *ng.ContainerRuntime != ContainerRuntimeDockerD && *ng.ContainerRuntime != ContainerRuntimeContainerD {
```

Reply:
I wanted to do something else in there. But I think I eventually did it somewhere else. :D thanks for the spot! :D